common/test/acceptance/tests/studio/test_studio_general.py

```diff
@@ -62,6 +71,145 @@ def test_dashboard_no_courses(self):
+        self.sign_up_page = SignupPage(self.browser)
+        self.login_page = LoginPage(self.browser)
+
+        self.course_outline_page = CourseOutlinePage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+        self.course_outline_sign_in_redirect_page = CourseOutlineSignInRedirectPage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+        self.course_fixture = CourseFixture(
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run'],
+            self.course_info['display_name'],
+        )
+        self.user = None
+
+    def install_course_fixture(self):
+        """
+        Install a course fixture
+        """
+        self.course_fixture.install()
+        self.user = self.course_fixture.user
+
+    def test_sign_up_from_home(self):
+        """
+        Scenario: Sign up from the homepage
+        Given I visit the Studio homepage
+        When I click the link with the text "Sign Up"
+        And I fill in the registration form
+        And I press the Create My Account button on the registration form
+        Then I should see an email verification prompt
+        """
+        # HowitworksPage is the home page when user
+        # is not logged in. Visit it.
+        home_page = HowitworksPage(self.browser)
+        home_page.visit()
+        # Click sign up
+        click_css(page=home_page, css='.action.action-signup', source_index=0, require_notification=False)
+        unique_number = uuid.uuid4().hex[:4]
+        registration_dic = {
+            '#email': '{}-email@host.com'.format(unique_number),
+            '#name': '{}-name'.format(unique_number),
+            '#username': '{}-username'.format(unique_number),
+            '#password': '{}-password'.format(unique_number),
+        }
+        self.sign_up_page.wait_for_page()
+        # Register the user.
+        self.sign_up_page.sign_up_user(registration_dic)
+        home = HomePage(self.browser)
+        home.wait_for_page()
+
+    def test_login_with_valid_redirect(self):
+        """
+        Scenario: Login with a valid redirect
+        Given I have opened a new course in Studio
+        And I am not logged in
+        And I visit the url "/course/slashes:MITx+999+Robot_Super_Course"
+        And I should see that the path is "/signin?next=/course/slashes%3AMITx%2B999%2BRobot_Super_Course"
+        When I fill in and submit the signin form
+        Then I should see that the path is "/course/slashes:MITx+999+Robot_Super_Course"
+        """
+        self.install_course_fixture()
+        # Get the url, browser should land here after sign in.
+        course_url = self.course_outline_sign_in_redirect_page.url
+        self.course_outline_sign_in_redirect_page.visit()
+        # Login
```

Comment:
Assert on the URL here to fulfill 'And I should see that the path is "/signin?next=/course/slashes%3AMITx%2B999%2BRobot_Super_Course"'